mistralrs-core/src/sequence.rs

```diff
@@ -231,10 +234,30 @@ impl Sequence {
         self.state.set(state);
     }
 
-    pub fn is_done(&self, tok: u32, eos_tok: u32, max_model_len: usize) -> Option<StopReason> {
+    pub fn generated_text(&self, tokenizer: &Tokenizer) -> String {
+        let token_ids = self
+            .logprobs()
+            .iter()
+            .map(|lp| lp.token)
+            .collect::<Vec<_>>();
+        let text = tokenizer
+            .decode(&token_ids, false)
+            .expect("Failed to decode tokens generated by model");
+
+        text
+    }
+
+    pub fn is_done(
+        &self,
+        tok: u32,
+        eos_tok: u32,
+        max_model_len: usize,
+        tokenizer: &Tokenizer,
+    ) -> Option<StopReason> {
         if tok == eos_tok {
             Some(StopReason::Eos)
         } else if self.stop_tokens.contains(&tok) {
+            // TODO: strip the stop token from the generated text?
```

Comment:
In general engines remove the stop token from text... But whatever, the downstream consumer can do it too easily.

Reply:
I'll create a different issue for this.